charts/kube-otel-stack/templates/collector.yaml

```diff
@@ -11,6 +11,10 @@ metadata:
     {{- toYaml . | nindent 4}}
   {{- end }}
   labels:
+    app: {{ $collectorName }}
+    {{- if $collector.opampManaged }}
+    opentelemetry.io/opamp-managed: "true"
```

Comment:
We should have a conditional for reporting or managed. I also think all of the collectors here should start as reporting

Reply:
how picky do you think we should be if someone tries to set both to true? add something here that will fail or let them create the collector?

Reply:
We should probably not let it happen in the chart

Reply:
afaict there doesn't seem to be a built in way to error on a given condition (most searches are pointing me to [sprig](https://masterminds.github.io/sprig/flow_control.html) which has a fail function, but i dont really want to add dependencies). any ideas how to achieve this?

Reply:
nah you should be able to just do an if else, give it some precedence. You can also do this with values.schema.json but that may be more work than its worth right now TBH.

Reply:
added if-else logic 👍

Reply:
in your pr you linked in the other thread, you have this 
```
    {{- if and $.Values.opAMPBridge.enabled $.Values.opAMPBridge.addReportingLabel }}
    opentelemetry.io/opamp-reporting: "true"
    {{- end }}
    {{- if and $.Values.opAMPBridge.enabled $.Values.opAMPBridge.addManagedLabel }}
    opentelemetry.io/opamp-managed: "true"
    {{- end }}
```

with the following in the bridge config
```
# Adds `opentelemetry.io/opamp-reporting: true` to all collectors
  addReportingLabel: true
  # Adds `opentelemetry.io/opamp-managed: true` to all collectors
  addManagedLabel: false
```

this all-or-nothing behavior feels a bit simpler for customers who we're prescribing these charts to. any qualms against me copying your work there instead of inlcuding a setting on each collector in values.yaml?

Reply:
nah go for it!

Reply:
maybe change it to an else if though :)